controller/src/main/java/io/pravega/controller/store/stream/ZKStream.java

```diff
@@ -309,12 +310,16 @@
     @Override
     CompletableFuture<VersionedMetadata<EpochRecord>> getCurrentEpochRecordData(boolean ignoreCached) {
         return getId().thenCompose(id -> {
+
+            CompletableFuture<VersionedMetadata<Integer>> future;
             if (ignoreCached) {
                 store.invalidateCache(currentEpochRecordPath, id);
+                future = store.getData(currentEpochRecordPath, x -> BitConverter.readInt(x, 0));
```

Comment:
Getting the data directly from the store when `ignoreCached` is true makes sense, but the semantics of `ignoreCached` is odd. `ignoreCached` is trying to make invalidate + repopulate atomic, but these two operations are really independent. My take is that `ignoreCached` should really be just that, *bypass the cache and get it from the store directly*. Invalidation should be performed separately when a new update happens.

Reply:
Yes, i was thinking of doing that. Ignore cached need not invalidate. I decided to keep the changes to a minimum, hence let it be there. I will remove the invalidate from these getters for ignoreCached part.